projects/angular/src/forms/datepicker/date-input-base.ts

```diff
@@ -264,33 +175,57 @@ export class ClrDateInput extends WrappedFormControl<ClrDateContainer> implement
     }
   }
 
-  private emitDateOutput(date: Date) {
-    if (!datesAreEqual(date, this.previousDateChange)) {
-      this.dateChange.emit(date);
-      this.previousDateChange = date;
-    } else if (!date && this.previousDateChange) {
-      this.dateChange.emit(null);
-      this.previousDateChange = null;
+  protected datepickerHasFormControl() {
+    return !!this.control;
+  }
+
+  /**
+   * Incase of date range error, both start & end date field valdiation has to be triggered
+   * if either of the field gets updated
+   */
+  protected validateDateRange() {
```

Comment:
I believe this belongs to the range specific children. They may have another common parent, that extends this class, to avoid code duplication between start and end, though.

Reply:
Yes, since both start date and end date component are using this kept in date-input-base component. Anyways I have wrapped it inside condition to check if its date range picker.